internal/testing/runchecks.sh

```diff
@@ -26,10 +26,41 @@ if [[ $# -gt 0 ]]; then
   exit 64
 fi
 
-result=0
+# The following logic lets us skip the (lengthy) installation process and tests
+# in some cases where the PR carries trivial changes that don't affect the code
+# (such as documentation-only).
+if [[ -z "$TRAVIS_BRANCH" || -z "$TRAVIS_PULL_REQUEST_SHA" ]]; then
+  echo "TRAVIS_BRANCH and TRAVIS_PULL_REQUEST_SHA environment variables must be set." 1>&2
+  exit 1
+fi
+
+tmpfile=$(mktemp)
+function cleanup() {
+  rm -rf "$tmpfile"
+}
+trap cleanup EXIT
+
+mergebase="$(git merge-base -- "$TRAVIS_BRANCH" "$TRAVIS_PULL_REQUEST_SHA")"
+git diff --name-only "$mergebase" "$TRAVIS_PULL_REQUEST_SHA" -- > $tmpfile
+
+# Find lines that don't start with internal/website in the diff log; if no such
+# lines are found, it means that we don't have to run tests. grep returns 1 in
+# this case.
+if grep -v ^internal/website $tmpfile; then
```

Comment:
Do you want to repeat this for only `.md` file PRs?

Reply:
I wanted to land the basic infrastructure first, and later we can add more patterns / exclusions as needed

Reply:
OK.